go/noise/laplace_noise_test.go

```diff
@@ -172,6 +172,496 @@ func TestDeltaForThresholdLaplace(t *testing.T) {
 		}
 	}
 }
+func TestInverseCDFLaplace(t *testing.T) {
+	for _, tc := range []struct {
+		desc               string
+		lambda, prob, want float64
+	}{
+		{
+			desc:   "Arbitrary test",
+			lambda: 4,
+			prob:   0.7875404240919761168041,
+			want:   3.4234254367,
+		},
+		{
+			desc:   "Arbitrary test",
+			lambda: 2,
+			prob:   0.1479685611330654517049,
+			want:   -2.435216544,
+		},
+		// For a probablity of 0.5 the result should be the zero regardless of lambda
+		{
+			desc:   "0.5 Probablity, output is zero",
+			lambda: 5,
+			prob:   0.5,
+			want:   0,
+		},
+		{
+			desc:   "0.5 Probablity, output is zero",
+			lambda: 10,
+			prob:   0.5,
+			want:   0,
+		},
+		// Tests for convergence to infinities with low and high probablities.
+		{
+			desc:   "Low probablity",
+			lambda: 3,
+			prob:   1.23757362E-15,
+			want:   -100.897429529251364,
+		},
+		{
+			desc:   "High probablitiy",
+			lambda: 3,
+			prob:   0.999999999876242638,
+			want:   66.3586531343406788,
+		},
+	} {
+		got := inverseCDFLaplace(tc.lambda, tc.prob)
+		if !approxEqual(got, tc.want) {
+			t.Errorf("TestInverseCDFLaplace(%f,%f)=%0.12f, want %0.12f, desc: %s", tc.lambda,
+				tc.prob, got, tc.want, tc.desc)
+		}
+	}
+
+}
+
+func TestConfidenceIntervalLaplace(t *testing.T) {
+	// Tests for computeConfidenceIntervalLaplace function
+	for _, tc := range []struct {
+		desc          string
+		noisedX       float64
+		lambda, alpha float64
+		want          ConfidenceInterval
+	}{
+		{
+			desc:    "Arbitrary test",
+			noisedX: 13,
+			lambda:  27.33333333333,
+			alpha:   0.05,
+			want:    ConfidenceInterval{-68.88334881, 94.88334881},
+		},
+		{
+			desc:    "Arbitrary test",
+			noisedX: 83.1235,
+			lambda:  60,
+			alpha:   0.24,
+			want:    ConfidenceInterval{-2.503481338, 168.7504813},
+		},
+		{
+			desc:    "Arbitrary test",
+			noisedX: 5,
+			lambda:  6.6666666666667,
+			alpha:   0.6,
+			want:    ConfidenceInterval{1.594495842, 8.405504158},
+		},
+		{
+			desc:    "Arbitrary test",
+			noisedX: 65.4621,
+			lambda:  700,
+			alpha:   0.8,
+			want:    ConfidenceInterval{-90.73838592, 221.6625859},
+		},
+		{
+			desc:    "Extremely low confidence level",
+			noisedX: 0,
+			lambda:  10,
+			alpha:   1 - 3.548957438e-10,
+			want:    ConfidenceInterval{-3.548957437370245055312e-9, 3.548957437370245055312e-9},
+		},
+		{
+			desc:    "Extremely high confidence level",
+			noisedX: 50,
+			lambda:  10,
+			alpha:   7.856382354e-10,
+			want:    ConfidenceInterval{-159.6452468975697118041, 259.6452468975697118041},
+		},
+	} {
+		got := computeConfidenceIntervalLaplace(tc.noisedX, tc.lambda, tc.alpha)
+		if !approxEqual(got.LowerBound, tc.want.LowerBound) {
+			t.Errorf("TestConfidenceIntervalLaplace(%f, %f, %f)=%0.10f, want %0.10f, desc %s, LowerBound is not equal",
```

Comment:
instead of TestConfidenceIntervalLaplace should be the method name which you test (the same for every other tests)

Reply:
Yeah I noticed that too but I thought TestComputeConfidenceIntervalLaplace was too long and you don't need to add Compute but I will edit it to follow this rule.